cpp/include/raft/linalg/detail/eig.cuh

```diff
@@ -91,6 +91,15 @@ void eigDC(raft::resources const& handle,
 #if CUDART_VERSION < 11010
   eigDC_legacy(handle, in, n_rows, n_cols, eig_vectors, eig_vals, stream);
 #else
+
+  // Use a new stream instead of `cudaStreamPerThread` to avoid cusolver bug # 4580093.
+  cudaStream_t stream_new;
+  cudaEvent_t sync_event;
+  RAFT_CUDA_TRY(cudaStreamCreate(&stream_new));
```

Comment:
Maybe use the `rmm::cuda_stream` instead (or a raft pool see below)?
We'd better never manually create/destroy resources: in an unlikely event someone catches the exception raised between creation of the stream here and its destruction at the end of the function, they'd get the leaked resource (cuda stream handle).